Doc/whatsnew/3.11.rst

```diff
@@ -1070,6 +1070,14 @@ Changes in the Python API
   before.
   (Contributed by Ma Lin in :issue:`35859`.)
 
+* More strict rules are now applied for numerical group references and
+  group names in regular expressions.
+  Only sequence of ASCII digits not starting with ``0`` (except group 0) is
+  now accepted as a numerical reference.
+  The group name in bytes patterns and replacement strings can now only
+  contain ASCII letters and digits and underscore.
+  (Contributed by Serhiy Storchaka in :issue:`91760`.)
```

Comment:
I think `:issue:` still ex[ects bpo ids.  I'll check if someone already created a `:gh:` role (there was some discussion about creating it), and create it if not).

Reply:
See #91888.

Reply:
The PR is now merged :)